Architecture.md

```diff
@@ -0,0 +1,100 @@
+- GameState -- a clean one for a new game, else a dirty one for a loaded save file.
+- userInterface -- an interface bridging the game to an arbitrary UI to describe it.
+
+### `game.js`
+
+Describes the game loop by organizing necessary dependencies and queuing the game actions in the passed `EventBus`.
+
+We use the setup logic in `game.js` to bind out `UI` and `gameState` to all of our `Rules` (see section below).
+
+### Outputs
+
+The game loop ends produces two outputs, one of which is arguable a byproduct.
+
+- `SIG_TERM` -- the game ends via a termination signal, outputting the associated code value.
+- `Save` -- the game outputs a save file (dirty `gameState`); this does _not_ terminate the game.
+
+## Game Loop Breakdown
+
+```javascript
+// Demonstrative code only to express high-level concepts
+const Rules = {
+    [ruleName: string]: [
+        ...actions: function ({ UI, notify }, gameState): void Event?
+    ]
+}
+
+while(eventBus.events && !SIG_TERM) {
+    let nextEvent = eventBus.next();
+    Rules[nextEvent.name].evaluate(nextEvent);
+}
+```
+
+### Rules Dictionary
+
+Game rules are written in the `Rules/index.js` file, which exists as a Dictionary with Keys of Rule Names and Values of an Array of Actions.
+
+An Action is a function with the signature `function ({ UI, notify }, gameState): void Event?`, where:
+
+- `UI` is the `userInterface` injected from `main.js`.
+- `notify` is a function to produce the `void Event?` effect (i.e., broadcast an event to evaluate another rule).
+- `gameState` is an object containing the game's state.
+
+The game loop runs on the assumption that named `Events` have a one-to-one association to a `Rule`.
+
+`Rule`s are stored in the dictionary (`Rules/index.js`) as a name with an ordered list of `Action`s that occur to evaluate the `Rule`. Think of it like steps take to follow or enact a rule. For example, a rule for moving a player piece around a board might be:
+
+```javascript
+const Rules = {
+    /* ... */
+    "MOVE_PLAYER": [
+        function promptCurrentPlayerToRollDice (_, gameState) { /* ... */ },
+        function movePlayerTokenSpacesEqualToDiceROll ({ UI, notify }, gameState) {
+            /* ... */
+            notify("PLAYER_MOVED");
+        }
+    ],
+    /* ... */
+}
+```
+
+It might be important to note that calls to `notify` usually only happen at the _end_ of a `Rule` (i.e., the last executed line of the last `Action` in the `Rule`'s array of `Action`s) and that the `EventBus` is synchronous, as mentioned earlier. This dictates the pace and order in which `Rule`s can be evaluated, so make sure you understand that well.
+
+`notify` has a return type of `void` but it broadcasts an `Event`, so we've presented the type signature `void Event` to denote this. You may have seen above for the `Action` signature that it returned `void Event?` - the `?` here simply denotes that not all actions emit `Event`s, hence it is an optional return type.
```

Comment:
`notify` has a return type of `void` but it broadcasts an `Event`. Because not all actions emit `Event`s, we denote the type signature `void Event?`.